### DISCUSSION TITLE
OMI glTF Working Group Meeting 2022-06-16

### DISCUSSION THREAD

**Discussion by fire** [omigroup/gltf-extensions#90] [General]
This meeting will be on 6/16/2022 at 10:30 PM (UTC) / 2:30 PM (PST) in the [OMI Discord](https://discord.gg/NJtT9grz5E) within the #omi-weekly-meeting channel. During the meeting, we will be using the #omi-gltf-extensions channel to manage a speaker queue, post links, and for any sidebar discussions.

To be notified of this meeting and others, subscribe to the [OMI Meetings and Events Calendar](https://calendar.google.com/calendar/u/1?cid=Y18wZHB1Z2Y5ZjgzZXE0cWVrbWI2b21xYmptZ0Bncm91cC5jYWxlbmRhci5nb29nbGUuY29t) or add yourself to the @omi-gltf-subgroup role in the #roles channel of the OMI Discord.

## Agenda:

- [KHR_audio](https://github.com/KhronosGroup/glTF/pull/2137) feedback
  - No new changes from the previous meeting.
  - Q? What is the contact on Khronos.
  - Q? What is the finish line?
  - Q? Who is on top of the topic at OMI?
  - Review PR from antpb
- [OMI_collider](https://github.com/omigroup/gltf-extensions/pull/63)
- OMI_id / [OMI_ref](https://github.com/omigroup/gltf-extensions/pull/85)
- End of the meeting show and tell period

To propose another item for the agenda, comment below (preferably before the meeting).

**Comment by madjin** [comment]
- to get the khr audio extension over the finish line we need:
  - to believe its ready, the ideas we attempt to get it right
  - setup / join meeting with khronos group to be ratified
  - what else can be marked as 'nice to have' vs 'must have'?

Lyuma:
There's an [open PR](https://github.com/omigroup/gltf-extensions/pull/88) in our group by antpb to adjust feedback from khronos

Reviewing recent comments:
![image](https://user-images.githubusercontent.com/32600939/174171463-08bcac81-507c-4e06-984d-11e756576723.png)
![image](https://user-images.githubusercontent.com/32600939/174171086-c76780d4-c07d-4dc2-bb38-94c69bfc64ca.png)

- Group checking to see if there's any addressable issues with it with a fine comb
- Robert working on implementation in Third Room by EOD

**Comment by madjin** [comment]
- I bring up janusweb [sound implementation](https://github.com/jbaicoianu/janusweb/blob/master/scripts/sound.js) as reference

![image](https://user-images.githubusercontent.com/32600939/174184597-ba21c660-8ec8-4843-bd5f-cc6db0e7f642.png)

asked bai some questions about it

*What's singleshot?*
> oh, it's so you can just fire a sound and it self-destructs when it's done playing
like...sometimes you want a sound that's persistent, like a looping sound - but sometimes you want something like a bounce, or some other reaction noise, and singleshot spawns the sound object at the give position, plays it once, and then dies
it's good for sounds that are going to play occasionally but not all the time - for sounds that happen frequently, like impact noises or weapon sound effects or whatever, it's often better to use an objectpool of sounds

*whats starttime used for usually?*
> I often use starttime for ambient loops - like that Blacktop Hoops world I built, it's based on a real streetball court in New York, and it has some ambient sound effect loops recorded from the real location, like 3-4 minutes long
so I can use that same ambient loop with spatialized sound emitters positioned around the court, on the sidewalk and streets, but with different offsets
so when you're moving around you get that sense of traffic and pedestrian and other random ambient noise coming from all around you, without having to use unique recordings for each location
> it's not really segmented, it's playing the whole loop at each location - but because it's offset by some random amount, it feels more like infinite non-repeating loops
> like you might hear a dog bark from your back left, and then a minute and 37 seconds later the same dog barks from your back right

*maybe that can be used for bird chirps*

> yeah exactly, that kind of thing
if you have a couple different ambient clips of differnet lengths you can pretty efficiently do what feels like infinitely-unique sounds, because they're gonna loop at different times
lots of tricks you can play to avoid the problem of super-repetitive sound....nothing drives me crazier than repeating ambient bird noises or whatever where it's a constant pattern 😄

> in the context of gltf suggestions I'd say, singleshot is pretty engine-specific, that's more of a runtime thing, whereas the starttime offset would actually be useful to be able to specify in a gltf, if gltf is going to support positional sound emitters that can reference the same audio buffer

```
janus sound.* => KHR_audio.*
     sound_id: .name
   singleshot: **MISSING**
   positional: .type=="positional"
         loop: .loop
    auto_play: .playing
    play_once: .loop==false?
         dist: .maxDistance
        pitch: **MISSING**
         gain: .gain
    starttime: **MISSING**
distancemodel: .distanceModel
      rolloff: .rolloffFactor
         rect: **MISSING**
```

Couple record player implementations (janusweb / vrchat udon)
- https://github.com/madjin/janus-custom-components/blob/master/media/recordplayer/recordplayer.js
- https://github.com/Vard-and-Temmie/Udon-Record-player

Couple new comments made to PRs
- https://github.com/omigroup/gltf-extensions/pull/63#issuecomment-1158180272
- https://github.com/omigroup/gltf-extensions/pull/88#issuecomment-1158176657
